ouroboros-consensus/src/Ouroboros/Consensus/Util/IOLike.hs

```diff
@@ -91,3 +93,17 @@ class ( MonadAsync              m
       ) => IOLike m where
 
 instance IOLike IO
+
+{-------------------------------------------------------------------------------
+  MonadDelay Wrappers
+-------------------------------------------------------------------------------}
+
+-- | This can be used to sleep for longer periods of time. 'threadDelay' can't
+-- be used for delays over ~35mins, because it uses an 'Int', which can
+-- overflow. See 'diffTimeToMicrosecondsAsInt' for more info.
+threadSleep :: MonadDelay m => DiffTime -> m ()
+threadSleep time = do
+    threadDelay maxDelay
+    when (maxDelay /= time) $ threadSleep (time - maxDelay)
+  where
+    maxDelay = min time $ fromIntegral (maxBound :: Int)
```

Comment:
I find it confusing that `maxDelay` is not really the maximum delay that we can wait for, both the minium of the maximum delay we can wait for and the requested wait.

What about:
```suggestion
threadSleep time
    | time > maxDelay
    = threadDelay maxDelay *> threadSleep (time - maxDelay)
    | otherwise
    = threadDelay time
  where
    maxDelay :: DiffTime
    maxDelay = fromIntegral (maxBound :: Int)
```